This adds emoji support to the word cloud generator

sontek opened the pull request.

amueller commented on setup.py:
```diff
@@ -11,5 +10,6 @@
     packages=['wordcloud'],
-    package_data={'wordcloud': ['stopwords', 'DroidSansMono.ttf']}
+    package_data={'wordcloud': ['stopwords', 'DroidSansMono.ttf']},
+    setup_requires=['fonttools']
```
Why?

sontek replied:
I needed fonttools package to detect which characters are supported in the font being used so we don't try to include characters that aren't supported.  For example, the font you ship with by default has no emojis in it but `DejaVu` has a lot

amueller commented on continuous_integration/install.sh:
```diff
@@ -40,7 +40,7 @@ elif [[ "$DISTRIB" == "ubuntu" ]]; then
 fi
 
-pip install -r requirements.txt
+pip install -e . -r requirements.txt
```
why that? We do the build in line 47.

amueller commented on wordcloud/wordcloud.py:
```diff
@@ -381,16 +403,41 @@ def process_text(self, text):
+        if not PY3 and type(text) is unicode or PY3:
+            flags = re.UNICODE
+
+        regex = re.compile(
```
this is pretty unclear. Can you explain?

sontek replied:
This is detecting the unicode range of emoji characters

amueller commented on wordcloud/wordcloud.py:
```diff
@@ -203,6 +208,23 @@ def __init__(self, font_path=None, width=400, height=200, margin=2,
+        )
+        supported_chars = []
+
+        for table in ttf["cmap"].tables:
```
you check whether it's in the font?

sontek replied:
Yeah, if someone says :poop: but the font doesn't have it, we skip it so we don't put `[]` in the word cloud

amueller replied:
I wouldn't put this in the init, but rather in the building of the word cloud. Also, it would be great to make this somehow optional so as not to get too many dependencies. It would also be great to factorize this functionality somewhat. Init and process text are already too long.

sontek replied:
I put it in init so you don't have the cost of scanning the font every time you want to process the text

amueller commented:
Please add documentation and test. Travis fails.
Also, I'd rather not add another dependency.

sontek commented:
@amueller I addressed most of your concerns.  I'm not sure how to make it pass on py27.  I haven't used 2.7 in a few years so I'm rusty on the differences

sontek commented:
@amueller This is ready for your review now I believe.

sontek commented:
![emoji cloud](https://cloud.githubusercontent.com/assets/151924/13004008/a4f00872-d12d-11e5-9868-55d23ae5d78d.png)

amueller commented:
that could looks pretty sweet, I have to say :)

amueller commented on setup.py:
```diff
@@ -11,5 +11,7 @@
     packages=['wordcloud'],
-    package_data={'wordcloud': ['stopwords', 'DroidSansMono.ttf']}
+    package_data={
+        'wordcloud': ['stopwords', 'DroidSansMono.ttf', 'Symbola.ttf']
```
have you checked the license?

amueller commented:
I''m a bit busy right now but I'll try to review soon. It's quite a bit of added complexity, though :-/
it would be nice if the changes could be more contained. I'm confused by conda not having fonttools. That's not a good sign, though :-/

sontek commented:
I don't have time to work on this anymore and it never got reviewed last year when I submitted it.  

Going to close it
